### PR TITLE
TFOpLambda layer support (Keras tensor)

### PR DESCRIPTION
**Pull request by negvet**
### Changes

Add support for the TFOpLambda layer, which is used when Keras tensor is enabled.

What is updated:
- Model converter. TFOpLambda serializes in a different way - config has its own structure, e.g. inbound_node format.
- Model transformer is getting updated as well to meet the new config of the model.
- Metatypes. op_names of op metatypes differ for TFOpLambda layer. Meta types are updated. Two new metatypes are added TFOpLambdaMetatype and TFSlicingOpLambdaMetatype. Metatype groups are updated as well.
- The main issues were faced with MASK-RCNN. The model itself is updated to be able to restore from config.

### Reason for changes

TFOpLambda layer is the only available wrapper layer for the tf and python ops (functional models) starting from TF 2.5.0 (no option to disable Keras tensor).

### Related tickets

71815, 71816

### Tests

Tests updated

### CODE REVIEW

File: nncf/tensorflow/graph/utils.py [hunk 1]
```diff
@@ -167,6 +168,58 @@ def _was_specially_wrapped_with_keras_export(layer, attr_name) -> bool:
...
+
+
+def get_list_level(lst: List) -> int:
+    return isinstance(lst, list) and list(map(get_list_level, lst or [0]))[0] + 1
+
+
+def check_oplambda_input_data(x: List) -> bool:
+    input_stracture = [type(item) for item in x]
+    return input_stracture in ([str, int, int], [str, int, int, dict])
+
```
Comment (alexsu52) [inline nncf/tensorflow/graph/utils.py:183]:
```suggestion
```

**Review by alexsu52 — COMMENTED** [review]
Please, provide e2e build results.

File: examples/tensorflow/segmentation/configs/quantization/mask_rcnn_coco_int8.json [hunk 2]
```diff
@@ -22,16 +22,37 @@
...
+        "mask_fcn_logits",
+        "conv5-mask",
+        "mask-conv-l3",
+        "mask-conv-l2",
+        "mask-conv-l1",
+        "mask-conv-l0",
+        "fc1",
+        "fc0",
+        "class-predict",
+        "box-predict",
```
[the remark this answers is not in the document]
Comment (negvet) [reply examples/tensorflow/segmentation/configs/quantization/mask_rcnn_coco_int8.json:55]: It will be updated

File: nncf/tensorflow/graph/pattern_operations.py [hunk 3]
```diff
@@ -47,6 +47,7 @@
 
 KERAS_ACTIVATIONS_OPERATIONS = {
     'type': ['ReLU',
+             'nn.relu',
```
[the remark this answers is not in the document]
Comment (negvet) [reply nncf/tensorflow/graph/pattern_operations.py]: Thanks, fixed!

File: nncf/tensorflow/graph/utils.py (hunk 1 above)
Comment (negvet) [reply nncf/tensorflow/graph/utils.py:183]: Thanks, fixed!

File: nncf/tensorflow/graph/metatypes/common.py [hunk 4]
```diff
@@ -109,7 +109,9 @@
 
 LAYER_METATYPES_AGNOSTIC_TO_DATA_PRECISION_WITH_MULTIPLE_INPUTS = [
     layer_metatypes.TFConcatenateLayerMetatype,
-    op_metatypes.TFConcatOpMetatype
+    op_metatypes.TFConcatOpMetatype,
+    op_metatypes.TFMaximumOpMetatype,
+    op_metatypes.TFMinimumOpMetatype,
```
[the remark this answers is not in the document]
Comment (negvet) [reply nncf/tensorflow/graph/metatypes/common.py:114]: That is correct. Simple option is to put `TFMaximumOpMetatype `and `TFMinimumOpMetatype` to the `LAYER_METATYPES_AGNOSTIC_TO_DATA_PRECISION_WITH_ONE_INPUT`. It will work, but actually, `maximum` and `minimum` has two inputs.

[the remark this answers is not in the document]
Comment (negvet) [reply nncf/tensorflow/graph/metatypes/common.py:114]: Another option is to create separate groups for `concat` and `max/min` ops. What do you think?

File: nncf/tensorflow/graph/converter.py [hunk 5]
```diff
@@ -672,7 +703,13 @@ def _get_multiple_input_layer_attributes(layer: tf.keras.layers.Layer) -> Multip
     else:
         input_shape = layer.input_shape
         output_shape = layer.output_shape
+
+        if not isinstance(input_shape, list):
```
[the remark this answers is not in the document]
Comment (negvet) [reply nncf/tensorflow/graph/converter.py:707]: As you correctly stated below, it is needed for `TFMaximumOpMetatype`, `TFMinimumOpMetatype`

File: nncf/tensorflow/graph/metatypes/common.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexsu52) [reply nncf/tensorflow/graph/metatypes/common.py:114]: I would prefer to create a separate group.

[the remark this answers is not in the document]
Comment (negvet) [reply nncf/tensorflow/graph/metatypes/common.py:117]: I created a separate group

File: nncf/tensorflow/graph/converter.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (negvet) [reply nncf/tensorflow/graph/converter.py]: `TFOpLambda` concat output_shape does not has type `List[List[int]]`, but `Tuple[int]`.
Have to keep
```
output_shape = layer.output_shape
if not isinstance(output_shape, list):
    output_shape = [output_shape]
```

File: examples/tensorflow/common/object_detection/ops/spatial_transform_ops.py [hunk 6]
```diff
@@ -11,6 +11,7 @@
  limitations under the License.
 """
 
+from tensorflow.keras import layers
```
Comment (andrey-churkin) [inline examples/tensorflow/common/object_detection/ops/spatial_transform_ops.py:14]: I'm not sure it is really necessary Why we can not use `tf.keras.layers`?

File: examples/tensorflow/segmentation/models/maskrcnn_model.py [hunk 7]
```diff
@@ -37,7 +36,7 @@ def _restore_baseline_weights(keras_model, checkpoint_path):
 
     assignment_map = {}
     for v in keras_model.variables:
-        var_name = v.name[v.name.find('/') + 1:] # Skip parent name
+        var_name = v.name
```
Comment (andrey-churkin) [inline examples/tensorflow/segmentation/models/maskrcnn_model.py:39]: Could you please confirm that the baseline checkpoint was successfully restored?

**Review by andrey-churkin — APPROVED** [review]

File: examples/tensorflow/segmentation/models/maskrcnn_model.py (hunk 7 above)
Comment (negvet) [reply examples/tensorflow/segmentation/models/maskrcnn_model.py:39]: Thanks for the notice! 
Yes, I confirm that the baseline checkpoint is restored.

File: examples/tensorflow/common/object_detection/ops/spatial_transform_ops.py (hunk 6 above)
Comment (negvet) [reply examples/tensorflow/common/object_detection/ops/spatial_transform_ops.py:14]: Sure, updated!

**Review by alexsu52 — CHANGES_REQUESTED** [review]
Please, update the results for new checkpoints in the readme.

**Review by vshampor — APPROVED** [review]
